daemon-tests/src/mocks/oracle.rs

```diff
@@ -30,41 +31,53 @@ impl xtra::Actor for OracleActor {
 
     async fn stopped(self) -> Self::Stop {}
 }
-impl Oracle for OracleActor {}
 
 #[xtra_productivity(message_impl = false)]
 impl OracleActor {
     async fn handle(
         &mut self,
-        msg: oracle::GetAnnouncement,
+        _msg: oracle::GetAnnouncement,
     ) -> Result<olivia::Announcement, oracle::NoAnnouncement> {
-        self.mock.lock().await.get_announcement(msg)
+        let announcement = self
+            .mock
+            .lock()
+            .await
+            .announcement
+            .clone()
+            .expect("announcement has been set");
+        Ok(announcement)
     }
 
-    async fn handle(&mut self, msg: oracle::MonitorAttestation) {
-        self.mock.lock().await.monitor_attestation(msg)
-    }
+    async fn handle(&mut self, _msg: oracle::MonitorAttestation) {}
 
-    async fn handle(&mut self, msg: oracle::Sync) {
-        self.mock.lock().await.sync(msg)
+    async fn handle(&mut self, _msg: oracle::NewAttestationFetched) -> anyhow::Result<()> {
+        Ok(())
     }
+
+    async fn handle(&mut self, _msg: oracle::Sync) {}
+}
+pub struct MockOracle {
+    executor: command::Executor,
+    announcement: Option<olivia::Announcement>,
 }
 
-#[automock]
-pub trait Oracle {
-    fn get_announcement(
-        &mut self,
-        _msg: oracle::GetAnnouncement,
-    ) -> Result<olivia::Announcement, oracle::NoAnnouncement> {
-        unreachable!("mockall will reimplement this method")
+impl MockOracle {
+    fn new(executor: command::Executor) -> Self {
+        Self {
+            executor,
+            announcement: None,
+        }
     }
 
-    fn monitor_attestation(&mut self, _msg: oracle::MonitorAttestation) {
-        unreachable!("mockall will reimplement this method")
+    pub async fn decrypt_cet(&mut self, id: OrderId, attestation: oracle::Attestation) {
```

Comment:
I find the name of this function a bit strange. Can we call this `simulate_attestation` instead?

Reply:
yep makes sense